TypeCobol.Analysis/Cfg/ControlFlowGraphBuilder.cs

```diff
@@ -0,0 +1,2695 @@
+        /// <param name="addToCurrentSentence">true if the block must be added to the current Sentence, false otherwise.</param>
+        /// <returns>The new Block</returns>
+        internal BasicBlockForNode CreateBlock(Node node, bool addToCurrentSentence)
+        {
+            var block = new BasicBlockForNode();
+            block.Index = this.CurrentProgramCfgBuilder.Cfg.AllBlocks.Count;
+            this.CurrentProgramCfgBuilder.Cfg.AllBlocks.Add(block);
+
+            if (node != null)
+            {
+                this.CurrentProgramCfgBuilder.Cfg.BlockFor[node] = block;
+                block.Instructions.AddLast(node);
+            }
+            if (addToCurrentSentence && this.CurrentProgramCfgBuilder.CurrentSentence != null)
+            {
+                this.CurrentProgramCfgBuilder.CurrentSentence.AddBlock(block);
+            }
+            if (CurrentDeclarativesContext != null)
+            {//This block is created in the context of a Declaratives.
+                block.SetFlag(BasicBlock<Node, D>.Flags.Declaratives, true);
+            }
+            return block;
+        }
+
+        /// <summary>
+        /// Group Index Counter
+        /// </summary>
+        private int GroupCounter = 0;
+        /// <summary>
+        /// Create a Group Basic Block for a node
+        /// </summary>
+        /// <param name="node">The leading node of the block</param>
+        /// <param name="addToCurrentSentence">true if the block must be added to the current Sentence, false otherwise.</param>
+        /// <returns>The new Block</returns>
+        internal BasicBlockForNodeGroup CreateGroupBlock(Node node, bool addToCurrentSentence)
+        {
+            var block = new BasicBlockForNodeGroup();
+            block.GroupIndex = ++GroupCounter;
+            block.Index = this.CurrentProgramCfgBuilder.Cfg.AllBlocks.Count;
+            this.CurrentProgramCfgBuilder.Cfg.AllBlocks.Add(block);
+            if (node != null)
+            {
+                this.CurrentProgramCfgBuilder.Cfg.BlockFor[node] = block;
+                block.Instructions.AddLast(node);
+            }
+            if (addToCurrentSentence && this.CurrentProgramCfgBuilder.CurrentSentence != null)
+            {
+                this.CurrentProgramCfgBuilder.CurrentSentence.AddBlock(block);
+            }
+            if (CurrentDeclarativesContext != null)
+            {//This group is created in the context of a Declaratives.
+                block.SetFlag(BasicBlock<Node, D>.Flags.Declaratives, true);
+            }
+            return block;
+        }
+
+        /// <summary>
+        /// Create a Fresh Control Flow Graph Builder.
+        /// </summary>
+        /// <returns>The fresh Control Flow Graph Builder</returns>
+        protected virtual ControlFlowGraphBuilder<D> CreateFreshControlFlowGraphBuilder(ControlFlowGraphBuilder<D> parentCfgBuilder = null)
+        {
+            return new ControlFlowGraphBuilder<D>();
```

Comment:
Bug, the parent is not used. this has been fixed in DFA though.

Reply:
There is another bug: see : 
https://github.com/TypeCobolTeam/TypeCobol/commit/68c7c477d08e8c5d86fed521be63a3f4ef290c7f

Reply:
done